lazy.go

```diff
@@ -51,7 +51,11 @@ type lazyConn struct {
 }
 
 func (l *lazyConn) Read(b []byte) (int, error) {
-	if !l.rhandshake {
+	l.rhlock.RLock()
+	rhandshake := l.rhandshake
+	l.rhlock.RUnlock()
```

Comment:
I was trying to do it previously with Once but I run into some problems. One of them was that I had to implement `AsyncDo` (simple) so not every time new goroutine is started when we want to run something once and then I got some lockup but I could not figure out what was causing it.